staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -747,3 +734,203 @@ func determineRequestedIP(svc *v1.Service, fwdRule *compute.ForwardingRule) stri
 
 	return ""
 }
+
+func getILBOptions(svc *v1.Service) ILBOptions {
+	return ILBOptions{AllowGlobalAccess: GetLoadBalancerAnnotationAllowGlobalAccess(svc)}
+}
+
+// forwardingRuleComposite is a composite type encapsulating both the GA and Beta ForwardingRules.
+// It exposes methods to compute the ForwardingRule object based on the given parameters and to compare 2 composite types
+// based on the version string.
+type forwardingRuleComposite struct {
+	allowGlobalAccess bool
+	name              string
+	description       *forwardingRuleDescription
+	ipAddress         string
+	backendService    string
+	ports             []string
+	ipProtocol        string
+	lbScheme          string
+	subnetwork        string
+	network           string
+}
+
+func (f *forwardingRuleComposite) Version() meta.Version {
+	return f.description.APIVersion
+}
+
+func (f *forwardingRuleComposite) Equal(other *forwardingRuleComposite) bool {
+	// IPAddress being an output-only field, equality check ignored if one is empty.
+	return (f.ipAddress == "" || other.ipAddress == "" || f.ipAddress == other.ipAddress) &&
+		f.ipProtocol == other.ipProtocol &&
+		f.lbScheme == other.lbScheme &&
+		equalStringSets(f.ports, other.ports) &&
+		f.backendService == other.backendService &&
+		f.allowGlobalAccess == other.allowGlobalAccess
+}
+
+// ToForwardingRuleComposite converts a compute beta or GA ForwardingRule into the composite type
+func ToForwardingRuleComposite(rule interface{}) (frc *forwardingRuleComposite, err error) {
+	switch v := rule.(type) {
+	case *compute.ForwardingRule:
```

Comment:
use a type switch https://tour.golang.org/methods/16:
```
switch fr := rule.(type) {
  case *compute.ForwardingRule:
    ...
  case *computebeta.ForwardingRule:
    ...
```

Reply:
This is already using type switch. Removed unnecessary typecast.